packages/server/src/router.ts

```diff
@@ -192,11 +202,36 @@ const defaultTransformer: CombinedDataTransformer = {
   output: { serialize: (obj) => obj, deserialize: (obj) => obj },
 };
 
+export type SwapProcedureContext<
+  TProcedure extends Procedure<any, any, any, any>,
+  TNewContext,
+> = TProcedure extends Procedure<
+  infer TInputContext,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  infer _TOldContext,
+  infer TInput,
+  infer TOutput
+>
+  ? Procedure<TInputContext, TNewContext, TInput, TOutput>
+  : never;
+
+export type SwapContext<
+  TObj extends ProcedureRecord<any, any, any, any>,
+  TNewContext,
+> = format<
+  {
+    [P in keyof TObj]: SwapProcedureContext<TObj[P], TNewContext>;
+  }
+>;
+
+/** @experimental The type signature of this class may change without warning. */
 export class Router<
```

Comment:
I've started using this https://api-extractor.com/pages/tsdoc/tag_internal/

Not sure what `@experimental` actually does 


```suggestion
/** 
 * @internal The type signature of this class may change without warning. 
 */
export class Router<
```